maven-repository-metadata/src/site/apt/index.apt

```diff
@@ -54,11 +54,3 @@ Maven Repository Metadata Model
    * {{{./apidocs/index.html}Java sources}} with Reader and Writers for the Xpp3 XML parser, to read and write <<<maven-metadata(-*).xml>>> files,
 
    * a {{{./repository-metadata.html}Descriptor Reference}}.
- 
- Notice: data about plugins in a directory representing a groupId is deprecated and will be removed in a future Maven version.
-~~ logic behind this:
-~~ 1. MNG-7266: maven-compat will be removed from future Maven version
-~~ 2. this will remove the code that updates plugins data: see MNG-7375/MPLUGIN-384 https://maven.apache.org/ref/3.8.4/maven-compat/apidocs/org/apache/maven/artifact/repository/metadata/GroupRepositoryMetadata.html
-~~ 3. this will lead to inconsistent data: removing it will be safer/more clear
-~~ but this logic still remains to be confirmed by clear consensus of the whole team
```

Comment:
IIUC, this is not wrong, but just too early, right ?

Reply:
No, this is plain wrong: assumes that two distinct things are "same thing" (maven-repository-metadata generated classes and maven-compat Metadata coming from Maven2, pre Aether times). 

It blurs by mistake two distinct things, while they are not same (but their purpose WAS same, but latter should not be used anymore, while former is and should be used by resolver-provider ONLY)

Reply:
can we clarify what fills the (effectively non-deprecated) metadata? And when Maven was fixed to update this metadata?
because this is the part that is not clear to me

Reply:
It was done on master (https://github.com/apache/maven/pull/555) and ported to maven-3.9.x (https://github.com/apache/maven/pull/691)

The code is (and should remain) confined in maven-resolver-provider, as it "hooks" into resolver https://github.com/apache/maven-resolver/blob/master/maven-resolver-impl/src/main/java/org/eclipse/aether/impl/MetadataGeneratorFactory.java and does the job by simply inspecting the payload. Also notice that all this code is package protected (again, as this happens by "hooking into resolver, and is not an API that one should tamper/call").